resources/packs/gcp/gcp.lr

```diff
@@ -552,42 +555,342 @@ private gcp.storage.bucket @defaults("id") {
   iamConfiguration dict
 }
 
-// GCP Cloud SQL
-gcp.sql {
-  //
-  instances() []gcp.sql.instance
+// GCP Cloud SQL Resources
+private gcp.project.sqlservices {
+  // Project ID
+  projectId string
+  // The list of Cloud SQL instances in the current project
+  instances() []gcp.project.sqlservices.instance
 }
 
 // GCP Cloud SQL Instance
-private gcp.sql.instance @defaults("name") {
-  // Instance name
-  name string
+private gcp.project.sqlservices.instance @defaults("name") {
+  // Internal ID
+  id string
+  // Project ID
+  projectId string
+  // All maintenance versions applicable on the instance
+  availableMaintenanceVersions []string
   // Backend type
   backendType string
-  // Connection name
+  // Connection name of the instance used in connection strings
   connectionName string
-  // Database version
+  // Creation timestamp
+  created time
+  // The current disk usage of the instance in bytes. This is deprecated, monitoring should be used instead
+  currentDiskSize int
+  // The current database version running on the instance
+  databaseInstalledVersion string
+  // The database engine type and version
   databaseVersion string
-  // GCE zone
+  // Disk encryption configuration
+  diskEncryptionConfiguration gcp.project.sqlservices.instance.diskEncryptionConfiguration
+  // Disk encryption status
+  diskEncryptionStatus gcp.project.sqlservices.instance.diskEncryptionStatus
+  // The name and status of the failover replica
+  failoverReplica gcp.project.sqlservices.instance.failoverReplica
+  // The Compute Engine zone that the instance is currently service from
   gceZone string
   // Instance type
   instanceType string
-  // Kind
-  kind string
-  // Current disk size
-  currentDiskSize int
-  // Maximum disk size
+  // The assigned IP addresses
+  ipAddresses []gcp.project.sqlservices.instance.ipMapping
+  // The current software version on the instance
+  maintenanceVersion string
+  // The name of the instance which will act as primary in the replica
+  masterInstanceName string
+  // The maximum disk size in bytes
   maxDiskSize int
-  // Instance state
-  state string
+  // Instance name
+  name string
   // Project
   project string // TODO: should this be id or name
   // Region
   region string
-  // Service account email address
-  serviceAccountEmailAddress string
+  // Replicas
+  replicaNames []string
   // Settings
   settings dict
+  // Service account email address
+  serviceAccountEmailAddress string
+  // Instance state
+  state string
+  // List of the databases in the current SQL instance
+  databases() []gcp.project.sqlservices.instance.database
+}
+
+// GCP Cloud SQL Instance database
+private gcp.project.sqlservices.instance.database @defaults("name") {
+  // Internal ID
```

Comment:
Similar to the other PRs we should remove all `Internal ID` since we do not need them

Reply:
I removed them from the higher level resources. The ones that are nested deep will require me to pass the ID of the parent all the way down and then return an ID based on it. That would be a lot of code, so I think for the deeply nested objects this approach is better